tests/Directory.Build.props

```diff
@@ -10,6 +10,16 @@
       CS1712: Type parameter 'type_parameter' has no matching typeparam tag in the XML comment on 'type_or_member' (but other type parameters do)
     -->
     <NoWarn>$(NoWarn);1573;1591;1712</NoWarn>
+
+    <IsHelixOnlyTestProject Condition="'$(IsHelixOnlyTestProject)' == ''">false</IsHelixOnlyTestProject>
+    <IsNonHelixOnlyTestProject Condition="'$(IsNonHelixOnlyTestProject)' == ''">false</IsNonHelixOnlyTestProject>
```

Comment:
Given the above is a boolean, do we actually need this property? Can't we just infer based on the above?

Reply:
We have 3 possible values - 1. helix only; 2. build only; 3. any . The two properties represent (1), and (2), and both being false means that (3) - no restrictions on where the tests can run.

We could use a string property with some fixed values like `['helix', 'non-helix', 'any']`, but then you have the value strings in multiple places where you want to check, and that could get misspelled and cause tests to get silently skipped.

I'm not attached to any specific way of doing it though.

The name isn't great either, for the second property - `IsNonHelixOnlyTestProject`. An alternative would be `IsBuildMachineOnlyTestProject` sounds less clear.

Reply:
In dotnet/aspnetcore, we have C# attributes to control whether a test runs on Helix or not. 

https://github.com/dotnet/aspnetcore/blob/515987ac92f801fb995d88a20282a61571523aab/src/Testing/src/xunit/SkipOnCIAttribute.cs
https://github.com/dotnet/aspnetcore/blob/515987ac92f801fb995d88a20282a61571523aab/src/Testing/src/xunit/SkipOnHelixAttribute.cs
https://github.com/dotnet/aspnetcore/blob/515987ac92f801fb995d88a20282a61571523aab/src/Testing/src/xunit/SkipNonHelix.cs

Would it be possible to do this that way? Then we wouldn't need to worry about misspelling, since it would be C# code.

Reply:
I see, yeah I think that as @eerhardt pointed out in a separate comment, I guess that the name is a bit confusing. I like his suggestion of dropping Only, or alternatively we can have something like `OnlyRunOnHelix` and `DoNotRunOnHelix` or something along those lines. Don't feel strongly about my suggestion haha, but I do think that the current version might be misleading.